src/core/unit-dependency-atom.c

```diff
@@ -0,0 +1,191 @@
+#include "unit-dependency-atom.h"
+
+static const UnitDependencyAtom atom_map[_UNIT_DEPENDENCY_MAX] = {
+        /* A table that maps high-level dependency types to low-level dependency "atoms". The latter actually
+         * describe specific facets of dependency behaviour. The former combine them into one user-facing
```

Comment:
Wouldn't it be clearer if the concept were to be called `UnitDependencyFacet`?

Reply:
well, naming things is always the hardest.

But I am pretty sure "atom" is the right name here, as we should emphasize that this is about splitting up dependency concepts into their basic, atomic building blocks. Hence "atom" sounds like the right word, since it nicely emphasizes that you can't split it up any further.

Reply:
Indeed. Reading the patch Atom was not very informative and the whole thing made sense only when I arrived at the bottom of it where the concept is defined. If the hunks would have been in the reverse order in the commit, I would have probably not questioned the name :)